components/dashboard/src/service/json-rpc-envvar-client.ts

```diff
@@ -0,0 +1,230 @@
+    DeleteUserEnvironmentVariableRequest,
+    DeleteUserEnvironmentVariableResponse,
+    EnvironmentVariableAdmission,
+    ListConfigurationEnvironmentVariablesRequest,
+    ListConfigurationEnvironmentVariablesResponse,
+    ListUserEnvironmentVariablesRequest,
+    ListUserEnvironmentVariablesResponse,
+    ResolveWorkspaceEnvironmentVariablesRequest,
+    ResolveWorkspaceEnvironmentVariablesResponse,
+    UpdateConfigurationEnvironmentVariableRequest,
+    UpdateConfigurationEnvironmentVariableResponse,
+    UpdateUserEnvironmentVariableRequest,
+    UpdateUserEnvironmentVariableResponse,
+} from "@gitpod/public-api/lib/gitpod/v1/envvar_pb";
+import { converter } from "./public-api";
+import { getGitpodService } from "./service";
+import { UserEnvVar, UserEnvVarValue } from "@gitpod/gitpod-protocol";
+
+export class JsonRpcEnvvarClient implements PromiseClient<typeof EnvironmentVariableService> {
+    async listUserEnvironmentVariables(
+        req: PartialMessage<ListUserEnvironmentVariablesRequest>,
+    ): Promise<ListUserEnvironmentVariablesResponse> {
+        const result = new ListUserEnvironmentVariablesResponse();
+        const userEnvVars = await getGitpodService().server.getAllEnvVars();
+        result.environmentVariables = userEnvVars.map((i) => converter.toUserEnvironmentVariable(i));
+
+        return result;
+    }
+
+    async updateUserEnvironmentVariable(
+        req: PartialMessage<UpdateUserEnvironmentVariableRequest>,
+    ): Promise<UpdateUserEnvironmentVariableResponse> {
+        if (!req.envVarId) {
+            throw new ConnectError("envVarId is required", Code.InvalidArgument);
+        }
+
+        const response = new UpdateUserEnvironmentVariableResponse();
+
+        const userEnvVars = await getGitpodService().server.getAllEnvVars();
+        const userEnvVarfound = userEnvVars.find((i) => i.id === req.envVarId);
+        if (userEnvVarfound) {
+            const variable: UserEnvVarValue = {
+                id: req.envVarId,
+                name: req.name ?? userEnvVarfound.name,
+                value: req.value ?? userEnvVarfound.value,
+                repositoryPattern: req.repositoryPattern ?? userEnvVarfound.repositoryPattern,
+            };
+            variable.repositoryPattern = UserEnvVar.normalizeRepoPattern(variable.repositoryPattern);
+
+            await getGitpodService().server.setEnvVar(variable);
+
+            const updatedUserEnvVars = await getGitpodService().server.getAllEnvVars();
+            const updatedUserEnvVar = updatedUserEnvVars.find((i) => i.id === req.envVarId);
+            if (!updatedUserEnvVar) {
+                throw new ConnectError("could not update env variable", Code.Internal);
+            }
+
+            response.environmentVariable = converter.toUserEnvironmentVariable(updatedUserEnvVar);
+            return response;
+        }
+
+        throw new ConnectError("env variable not found", Code.InvalidArgument);
```

Comment:
I have one thing not so sure, that should `not found` use `Code.NotFound`? 

But it makes sense to use `Code.InvalidArgument` too since it shouldn't reach this function without UX.

Not blocker but want to align us all. cc @akosyakov @jeanp413

Reply:
should be not found, I remember changing it 🤔 , will fix it

Reply:
please use ApplicationError as agreed today, drop ConnectError, only public-api-converter should translate them, no usage in any other places

Reply:
we clean up today other places: https://github.com/gitpod-io/gitpod/pull/19092

Reply:
@akosyakov  Maybe rebase on your PR then do it together here?

Reply:
Not sure what you mean, mentioned PR already merged.

It is rather easy to fix, open json and grpc impl files, do
```
search: ConnectError\((".*"), Code.InvalidArgument\);
replace: ApplicationError(ErrorCodes.BAD_REQUEST, $1);
```

then do the same for NOT_FOUND